### PR TITLE
Update Nesterov implementation docs

### LINKED ISSUE #19899: Documentation feature request: Explain Nesterov Accelerated Gradient implementation
I haven't filled out the provided form because this is a feature request for the documentation.

The documentation for [`tf.train.MomentumOptimizer`](https://www.tensorflow.org/api_docs/python/tf/train/MomentumOptimizer) offers a `use_nesterov` parameter on which the documentation says the following:

> `use_nesterov`: If True use Nesterov Momentum. See [Sutskever et al.,
> 2013](http://proceedings.mlr.press/v28/sutskever13.pdf). This
> implementation always computes gradients at the value of the
> variable(s) passed to the optimizer. Using Nesterov Momentum makes the
> variable(s) track the values called `theta_t + mu*v_t` in the paper.

The problem is that the linked paper outlines the normal NAG algorithm, which requires computation of a gradient at a different value to those provided (namely at the next step). This is not clarified in the documentation and led me to some confusion. I ended up [asking this question on StackOverflow](https://stackoverflow.com/questions/50774683/how-is-nesterovs-accelerated-gradient-descent-implemented-in-tensorflow) and cobbled together [an answer myself](https://stackoverflow.com/a/50774886/1613983), however I think the answer by `user1735003` is so complete that without too much wrangling it could greatly enhance the documentation:

[Answer by `user1735003`](https://stackoverflow.com/a/50778921/1613983)

Some clarification to the fact that tensorflow actually implements a modified version of the algorithm which is only correct under certain conditions would have been very helpful and would have saved me some time.

### PR DESCRIPTION
**Pull request by accraze**
Clarification that this is a modified version of the algorithm which is only correct under certain conditions.

Fixes #19899

### CODE REVIEW

**Review by martinwicke — DISMISSED** [review]

**Comment by martinwicke** [comment]
@accraze could you make sure that the lines are hard-wrapped to 80 chars and that there are no trailing spaces? Thanks!

**Comment by accraze** [comment]
@martinwicke I fixed the line lengths, but it seems to have dismissed your review.
